src/main/resources/plugins/dataset/TestbotPlugin.py

```diff
@@ -0,0 +1,546 @@
+            name_node_ip = socket.gethostbyname(self.cluster_name + '-' + 'hadoop-mgr-1')
+            name_node_url = 'http://' + name_node_ip + ':50070' + '/webhdfs/v1/user/pnda?user.name=pnda&' \
+                                                                  'op=GETFILESTATUS'
+            name_node_response = requests.get(name_node_url)
+            if name_node_response.status_code == 200:
+                self.hdfs_url = 'http://' + name_node_ip + ':50070'
+                name_node_status = True
+                LOGGER.debug("Found Ip address of name_node(hadoop-mgr-1) where webhdfs is running")
+            # Get second NAME_NODE IP address if the response from the first NAME NODE is 403
+            elif name_node_response.status_code == 403:
+                name_node_ip = socket.gethostbyname(self.cluster_name + '-' + 'hadoop-mgr-2')
+                name_node_url = 'http://' + name_node_ip + ':50070' + '/webhdfs/v1/user/pnda?user.name=pnda&' \
+                                                                      'op=GETFILESTATUS'
+                new_node_response = requests.get(name_node_url)
+                if new_node_response.status_code is 200:
+                    self.hdfs_url = 'http://' + name_node_ip + ':50070'
+                    name_node_status = True
+                    LOGGER.debug("Found Ip address of name_node(hadoop-mgr-2) where webhdfs is running")
+                else:
+                    LOGGER.warning("Couldn't get any response from hadoop-mgr-2 node")
+            else:
+                LOGGER.warning("Couldn't get response from hadoop-mgr-1 node")
+        except Exception as e:
+            LOGGER.error("Exception while executing get_name_node due to %s", e)
+        return name_node_status
+
+    # Check whether file has been modified recently with in the given time
+    def get_latest_modified_file(self, dataset_data):
+        avro_file_name = ''
+        try:
+            response_dict = json.loads(dataset_data.text)
+            file_data = response_dict['FileStatuses']['FileStatus']
+            if file_data:
+                LOGGER.debug("Found file data regarding the given dataset")
+                for avro_file in file_data:
+                    # Calculating current_time, cron interval time in milliseconds
+                    current_milli_time = int(time.time() * 1000)
+                    ref_time = int(current_milli_time - (self.cron_interval * 60000))
+                    modified_milli_time = avro_file.get('modificationTime')
+
+                    # Checking whether the modified time lies in between before_time, now_time.
+                    if modified_milli_time in range(ref_time, current_milli_time):
+                        LOGGER.debug("File has been modified recently with in last cron interval time")
+                        avro_file_name = avro_file.get('pathSuffix')
+                        return avro_file_name
```

Comment:
not keen on returns like this out of the middle of a function without a good reason to do so

Reply:
Sure. Will make required changes for all the comments provided.
Thank you!